### PR TITLE
NIC Bridged: Allow use of security.ipv6_filtering when no IPv6 allocation available

### PR DESCRIPTION
**Pull request by tomponline**

### CODE REVIEW

File: lxd/device/nic_bridged.go [hunk 1]
```diff
@@ -612,7 +619,7 @@ func (d *nicBridged) allocateFilterIPs(n *network.Network) (net.IP, net.IP, erro
 
 	// Check DHCPv6 is enabled on parent if dynamic IPv6 allocation is needed.
 	if shared.IsTrue(d.config["security.ipv6_filtering"]) && IPv6 == nil && !canIPv6Allocate {
-		return nil, nil, fmt.Errorf("Cannot use security.ipv6_filtering as DHCPv6 is disabled or no IPv6 on parent %s and no static IPv6 address set", d.config["parent"])
+		logger.Warn("IPv6 filtering enabled without a static IPv6 specified or dynamic allocation enabled, all IPv6 traffic will be blocked", log.Ctx{"project": d.inst.Project(), "instance": d.inst.Name(), "dev": d.name})
```
Comment (stgraber) [inline lxd/device/nic_bridged.go]: I'm tempted to say this should go to Debug if not entirely removed.
Environments (thinking of Travis) that run thousands of containers with such a setup will get tired of the warning on every startup :)

Comment (tomponline) [reply lxd/device/nic_bridged.go]: @stgraber ok will remove.

File: lxd/device/nic_bridged.go [hunk 2]
```diff
@@ -614,7 +622,7 @@ func (d *nicBridged) allocateFilterIPs(n *network.Network) (net.IP, net.IP, erro
 
 	// Check DHCPv4 is enabled on parent if dynamic IPv4 allocation is needed.
 	if shared.IsTrue(d.config["security.ipv4_filtering"]) && IPv4 == nil && !canIPv4Allocate {
-		return nil, nil, fmt.Errorf("Cannot use security.ipv4_filtering as DHCPv4 is disabled or no IPv4 on parent %s and no static IPv4 address set", d.config["parent"])
+		logger.Warn("IPv4 filtering enabled without a static IPv4 specified or dynamic allocation enabled, all IPv4 and ARP traffic will be blocked", log.Ctx{"project": d.inst.Project(), "instance": d.inst.Name(), "dev": d.name})
```
Comment (stgraber) [inline lxd/device/nic_bridged.go]: Same comment as for IPv6

**Comment by tomponline** [comment]
@stgraber made those changes.

**Pull request merged**